mindsdb/libs/data_sources/file_ds.py

```diff
@@ -183,23 +180,16 @@ def _setup(self,file, clean_rows = True, custom_parser = None):
             file_data = df.values.tolist()
         
         else:
-            raise ValueError('Could not load file into any format, supported formats are csv, json, xls, xslx')
-
-        for col in header:
-            col_map[col] = col
+            raise ValueError('Could not load file into any format, supported formats are csv, json, xls, xlsx')
 
         if clean_rows == True:
-            file_list_data = []
-            for row in file_data:
-                row = self.cleanRow(row)
-                file_list_data.append(row)
+            file_list_data = [self.cleanRow(row) for row in file_data]
         else:
             file_list_data = file_data
 
+        col_map = dict((col, col) for col in header)
+
         try:
             return pd.DataFrame(file_list_data, columns=header), col_map
-        except:
-            with open(file) as f:
-                f.readline()    # header
-                dialect = csv.Sniffer().sniff(f.readline())
```

Comment:
Is this bit of code no longer required ?

I.e. does the sniffing in `_getDataIo` work the exact same way thus making this one redundant ?

Reply:
since we're calling pd.read_csv, we're assuming it's a csv file, therefore dialect must be defined as it's always returned by `_getDataIo` in case of a csv file (https://github.com/mindsdb/mindsdb/blob/master/mindsdb/libs/data_sources/file_ds.py#L138)

Reply:
Fair point, please bump the version in __about__.py and merge

Reply:
Will do